src/DataValues/DecimalValue.php

```diff
@@ -93,33 +85,25 @@ private function assertNumberString( $number ) {
 	 * @throws InvalidArgumentException
 	 */
 	private function convertToDecimal( $number ) {
-		if ( !is_int( $number ) && !is_float( $number ) ) {
-			throw new InvalidArgumentException( '$number must be an int or float.' );
-		}
```

Comment:
This is done two times, literally one line of code before this method is called.